nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -1003,9 +1007,14 @@ private boolean overriddenMethodReturnsNonNull(
     // In JSpecify mode, for generic methods, we additionally need to check the return nullness
     // using the type parameters from the type enclosing the overriding method
     if (config.isJSpecifyMode()) {
-      return GenericsChecks.getGenericMethodReturnTypeNullness(
-              overriddenMethod, enclosingSymbol, state, config)
-          .equals(Nullness.NONNULL);
+
+      return (memberReferenceTree != null)
+          ? GenericsChecks.getGenericMethodReturnTypeNullness(
+                  overriddenMethod, ASTHelpers.getType(memberReferenceTree), state, config)
+              .equals(Nullness.NONNULL)
+          : GenericsChecks.getGenericMethodReturnTypeNullness(
+                  overriddenMethod, enclosingSymbol, state, config)
+              .equals(Nullness.NONNULL);
```

Comment:
Can we rewrite this as an if-then-else similar to what we did in #845, with a comment to document each case?

Reply:
Yes, Will do!